coding/javascript.md

```diff
@@ -37,6 +37,26 @@ We use [yarn](https://yarnpkg.com/en) for dependency management.
 
 We use the [jest](https://jestjs.io) unittest framework.
 
+#### General Testing Principals
+  * Avoid setting up state in `beforeEach`. It should be clear from reading a test in isolation what state it is in.
+  * Use test factories to reduce boilerplate for commonly instantiated objects.
+  * Don't test external code/libraries.
+
+#### Testing React
```

Comment:
When testing the output of a component we've found it helpful to test small changes (is a string correct, does something have a specific class, does an element exist/not exist in the output) with Enzyme vs doing a whole snapshot. This way it is clear to the reviewer/future coder what is really being tested. Then snapshots can be used for checking the full output where required.

Maybe that's covered in your notes before, but might be helpful to spell out when to use Enzyme and when to use snapshots.